packages/shared_preferences/shared_preferences_ios/pigeons/messages.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:pigeon/pigeon.dart';
+
+@ConfigurePigeon(PigeonOptions(
+  dartOut: 'lib/messages.g.dart',
+  objcHeaderOut: 'ios/Classes/messages.g.h',
+  objcSourceOut: 'ios/Classes/messages.g.m',
+  copyrightHeader: 'pigeons/copyright_header.txt',
+))
+@HostApi()
+abstract class SharedPreferencesApi {
+  bool remove(String key);
+  bool setBool(String key, bool value);
+  bool setDouble(String key, double value);
+  bool setInt(String key, int value);
+  bool setString(String key, String value);
+  bool setStringList(String key, List<String> value);
```

Comment:
I think it's better to match the Dart interface as closely as possible instead instead of trying to be smart in the glue code (as per other comment).

Reply:
[You do?](https://github.com/flutter/plugins/pull/4732/files#diff-a0d2163c6e027d74e130d1dda3b776bec03fc4019398abb625864b3f9d1cd841R14)

Reply:
And specifically, my point here is not just that you are putting logic in the Dart code there, but the purpose of that logic is to maximally fragment the calls by value type. You've created an API later that not only has more calls than the underlying host API (which is what I suggest matching), but far more than the Dart interface. Your argument against reducing the number from five setters to the is that you want to better match the side that only has one...

Reply:
Yep, I'd say that's matching the Dart interface.  The interface is weird in that case because it's defined in the comments and in the code not just in code because Dart lacks the ability to express it in code.  There is no logic in the dart file, just dispatching.

Reply:
I know it's late there, we can hash out the last mile of this conversation on Monday, over vc if we have to.